app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,37 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: %i(create destroy)
+  before_action :correct_user, only: :destroy
+  
+  def create; 
+    @micropost = current_user.microposts.build micropost_params
+    @micropost.image.attach params[:micropost][:image]
+    if @micropost.save
+      flash[:success] = "Micropost created!"
+      redirect_to root_url
+    else
+      @feed_items = current_user.feed.paginate page: params[:page]
+      render "static_pages/home"
+    end
+  end
+  
+  def destroy
+    if @micropost.destroy
+      flash[:success] = "Micropost deleted"
+      redirect_to request.referrer || root_url
+    else
+      flash[:danger] = "Micropost not deleted"
+      redirect_to root_url
+    end
+  end
+  
+  private
+  
+  def micropost_params
+      params.require(:micropost).permit :content, :image
+  end
+  
+  def correct_user
+      @micropost = current_user.microposts.find_by id: params[:id]
```

Comment:
please check indent

Reply:
Can you please guide me? I don't understand

Reply:
Done